graylog2-web-interface/src/components/sidecars/configuration-forms/EditConfigurationVariableModal.jsx

```diff
@@ -0,0 +1,146 @@
+import PropTypes from 'prop-types';
+import lodash from 'lodash';
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import { BootstrapModalForm, Input } from 'components/bootstrap';
+import CombinedProvider from 'injection/CombinedProvider';
+import ConfigurationHelperStyle from './ConfigurationHelper.css';
+
+const { ConfigurationVariableActions } = CombinedProvider.get('ConfigurationVariable');
+
+class EditConfigurationVariableModal extends React.Component {
+  static propTypes = {
+    create: PropTypes.boolean,
+    id: PropTypes.string,
+    name: PropTypes.string,
+    description: PropTypes.string,
+    content: PropTypes.string,
+    saveConfigurationVariable: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    create: false,
+    id: '',
+    name: '',
+    description: '',
+    content: '',
+  };
+
+  state = {
+    id: this.props.id,
+    name: this.props.name,
+    description: this.props.description,
+    content: this.props.content,
+    errors: {},
+  };
+
+  _hasErrors = () => {
+    return !lodash.isEmpty(this.state.errors);
+  };
+
+  openModal = () => {
+    this.modal.open();
+  };
+
+  _getId = (prefixIdName) => {
+    return prefixIdName + this.state.name;
+  };
+
+  _closeModal = () => {
+    this.modal.close();
```

Comment:
We should reset the modal contents after closing it, at least for consistency with other modals in the product.